src/network/linkers_socket.cpp

```diff
@@ -186,7 +187,9 @@ void Linkers::Construct() {
   listener_->SetTimeout(socket_timeout_);
   listener_->Listen(incoming_cnt);
   std::thread listen_thread(&Linkers::ListenThread, this, incoming_cnt);
-  const int connect_fail_retry_cnt = 20;
+  const int connect_fail_retries_factor_machine = 25;
+  const int connect_fail_retries_scale_factor = static_cast<int>(num_machines_ / connect_fail_retries_factor_machine);
+  const int connect_fail_retry_cnt = std::max(20, connect_fail_retries_scale_factor);
```

Comment:
so it is 20 when `num_machines_ = 500`, Is this too small?

Reply:
it's 20 for `num_machines_` less than 500, correct. It's too small if we're spinning say > 1000 `num_machines_`, in which case time to spin them up and get them running itself takes a while ( especially when we launch them as containers say in kubernetes cluster )